### DISCUSSION TITLE
6.2 > Added new uitype Barcode

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#14942] [General] [answered]
Hi, release notes says" Added new uitype Barcode, but I cannot figure out how to understand it. 
Appreciate some details

Best

**Comment by rskrzypczak** [comment, marked as answer]
The new barcode field type has just been pre-added.
At the moment it cannot be added from the GUI.
It has not been fully tested yet.
If you are a developer and would like to test it, you have to add it manually.
[Example](https://user-images.githubusercontent.com/3193498/101043159-4b469380-357e-11eb-9399-57ba533c6a71.png)

**Reply by IanMuller** [reply]
Looks interesting. Hoping it will support QR Codes 

Best

**Comment by konadrian** [comment]
Yes it support QR Code for barcodes we use https://github.com/milon/barcode so you can add any of the types (not all of them were test)
So to add this uitype you need to update uitype  and fieldparams in vtiger_field table 
Example value for field params column 
`{"barcodeDisplayType":"barcodeAndValue","barcodeType":"QRCODE","barcodeClass":"DNS2D","width":10,"height":10}`

![image](https://user-images.githubusercontent.com/3193498/125186200-307d8100-e229-11eb-90c7-be27c811fcce.png)

**Reply by TonyM15** [reply]
What variable should we use in PDF/Email template to get image instead of linked text?

**Comment by mariuszkrzaczkowski** [comment]
```
$(barcode : type=EAN13 class=DNS1D , value=12345678)$
$(barcode : type=EAN13 class=DNS1D , fieldName=number)$
```

**Reply by TonyM15** [reply]
Not working in PDF templates :/

**Comment by mariuszkrzaczkowski** [comment]
give sample html of pdf template and field value

**Reply by TonyM15** [reply]
`<pre>
$(barcode : type=EAN13 class=DNS1D , value=12345678)$</pre>

<p style="padding-top:70px">&nbsp;</p>

<table align="right" border="0" cellpadding="1" cellspacing="1" style="width:500px">
	<tbody>
		<tr>
			<td>
			<p><span style="font-family:'DejaVu Sans'"><span style="font-size:10px">Account:<strong> $(record : accountname)$</strong></span></span><br />
			<span style="font-family:'DejaVu Sans'; font-size:10px">Type: $(record : relationtype)$</span><br />
			<span><span style="font-size:10px">Record ID:07000$(record : RecordId)$</span></span></p>
			</td>
		</tr>
	</tbody>
</table>
`

**Reply by TonyM15** [reply]
It returns:
Argument 1 passed to YetiForcePDF\Style\NumericValue::setUnit() must be of the type string, null given, called in /var/www/html/brd/vendor/yetiforce/yetiforcepdf/lib/Style/Normalizer/Normalizer.php on line 100

**Reply by mariuszkrzaczkowski** [reply]
what version of crm system?

**Reply by TonyM15** [reply]
Marius, found issue, pre tag O.o sorry for alarm

**Reply by garryatbrighton** [reply]
last stable

**Comment by garryatbrighton** [comment]
I gave a try, and not sure but it not printing out "qr-code" field value as an image. WHat exactly syntax should be used for printing image from field?
